### PR TITLE
Inherit attributes from Restart{Source,Flow,Sink}

### PR DESCRIPTION
**Pull request by armanbilge**
References #24810

This PR enables inner operators to inherit attributes from a surrounding restart operator.

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

**Review by johanandren — APPROVED** [review]

**Comment by johanandren** [comment]
Would be good with test coverage verifying that it works and protecting against regression. 

Can be done by writing a custom graph stage in the test picking up attributes and handing them to the test for assertion. You can look at the small custom graph stages in  `akka.stream.scaladsl.AttributesSpec` to see how to do that.

**Comment by armanbilge** [comment]
Sounds good, I'll give that a shot!

**Comment by armanbilge** [comment]
Done. It's somewhat uglier than `akka.stream.scaladsl.AttributesSpec` because the Restart operators don't give access to materialized values of their inner operators, but I hope it's okay.

**Review by johanandren — APPROVED** [review]
Great, LGTM, thanks!

**Comment by armanbilge** [comment]
Sorry to see that this missed the last release. Any chance it might make the next?

**Comment by johanandren** [comment]
Only waiting for a second core team review before we can merge, so that should be possible.

**Review by chbatey — APPROVED** [review]

**Comment by armanbilge** [comment]
Thank you both!

**Pull request merged**
